File: vnHealthyCare/app/src/main/java/com/giangnb/pc_covid_clone/features/home/HomeFragment.java
package com.giangnb.pc_covid_clone.features.home;

import android.Manifest;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;

import com.giangnb.pc_covid_clone.R;
import com.giangnb.pc_covid_clone.databinding.FragmentHomeBinding;
import com.giangnb.pc_covid_clone.features.qrcode.QrCodeScannerFragment;

public class HomeFragment extends Fragment
        implements HomeContract.View {

    private int REQUEST_CODE_CAM = 111;
    private FragmentHomeBinding binding;
    private HomeViewModel viewModel = new HomeViewModel();

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = FragmentHomeBinding.inflate(inflater,container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        initListenerOnClick();
    }

    @Override
    public void initListenerOnClick() {
        binding.btnMenu.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.menuFragment);
            }
        });

        binding.btnNotify.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.notifyFragment);
            }
        });

        binding.btnQrCodeManagement.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.qrManagementFragment);
            }
        });

        binding.btnMedicalDeclaration.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.medicalDeclarationFragment);
            }
        });

        binding.btnReport.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).navigate(R.id.reportFragment);
            }
        });

        binding.btnQrScanner.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                requestPermissionCAM();
                Navigation.findNavController(view).navigate(R.id.qrCodeScannerFragment);
            }
        });
    }

    private void requestPermissionCAM() {
        if (ContextCompat.checkSelfPermission(getContext(),
                Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(getActivity(),
                    new String[]{Manifest.permission.CAMERA}, REQUEST_CODE_CAM);
        }
    }
}

File: DataStorage/src/main/java/com/data/storages/commons/ui/home/HomeFragment.kt
package com.data.storages.commons.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import com.data.storages.commons.ui.home.HomeItem.Companion.ITEM_HOME_CAMERA_ID
import com.data.storages.commons.ui.home.HomeItem.Companion.ITEM_HOME_CLOUD_STORAGE_ID
import com.data.storages.commons.ui.home.HomeItem.Companion.ITEM_HOME_EXTERNAL_STORAGE_ID
import com.data.storages.commons.ui.home.HomeItem.Companion.ITEM_HOME_INTERNAL_STORAGE_ID
import com.data.storages.commons.ui.home.HomeItem.Companion.ITEM_HOME_REALTIME_DATABASE_ID
import com.data.storages.commons.ui.home.HomeItem.Companion.ITEM_HOME_ROOM_DATABASE_ID
import com.data.storages.databinding.FragmentHomeBinding

class HomeFragment : Fragment(), HomeAdapter.IHomeCallBack {
    private var _binding: FragmentHomeBinding? = null
    private val binding get() = _binding!!
    private lateinit var viewModel: HomeViewModel
    private lateinit var adapter: HomeAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        viewModel = ViewModelProvider(this)[HomeViewModel::class.java]
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentHomeBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        adapter = HomeAdapter(viewModel.initializeHomeItems(), this)
        binding.recyclerview.adapter = adapter
    }

    override fun onItemClick(item: HomeItem) {
        when (item.id) {
            ITEM_HOME_CAMERA_ID -> {
                Toast.makeText(requireContext(), ITEM_HOME_CAMERA_ID, Toast.LENGTH_SHORT).show()
            }

            ITEM_HOME_INTERNAL_STORAGE_ID -> {
                Toast.makeText(requireContext(), ITEM_HOME_INTERNAL_STORAGE_ID, Toast.LENGTH_SHORT)
                    .show()
            }

            ITEM_HOME_EXTERNAL_STORAGE_ID -> {
                Toast.makeText(requireContext(), ITEM_HOME_EXTERNAL_STORAGE_ID, Toast.LENGTH_SHORT)
                    .show()
            }

            ITEM_HOME_CLOUD_STORAGE_ID -> {
                Toast.makeText(requireContext(), ITEM_HOME_CLOUD_STORAGE_ID, Toast.LENGTH_SHORT)
                    .show()
            }

            ITEM_HOME_REALTIME_DATABASE_ID -> {
                Toast.makeText(requireContext(), ITEM_HOME_REALTIME_DATABASE_ID, Toast.LENGTH_SHORT)
                    .show()
            }

            ITEM_HOME_ROOM_DATABASE_ID -> {
                Toast.makeText(requireContext(), ITEM_HOME_ROOM_DATABASE_ID, Toast.LENGTH_SHORT)
                    .show()
            }
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }
}